src/taipy/core/exceptions/exceptions.py

```diff
@@ -147,10 +150,17 @@ def __init__(self, pipeline_id: str, scenario_id: str):
 
 
 class PipelineTaskDoesNotExistInSameScenario(Exception):
-    """Raised if the Task of a Pipeline does not exist in the same Scenario."""
+    """Raised if the Tasks of a Pipeline does not exist in the same Scenario."""
+
+    def __init__(self, task_ids: List[Union[str, None]], pipeline_name: str, scenario_id: str):
```

Comment:
```suggestion
    def __init__(self, task_ids: List[str], pipeline_name: str, scenario_id: str):
```
The type hint here is weird, it should not be a List of None. Is this what you meant?

Reply:
ah you're right, I think it somehow got messed up with some of the automation tool for type hint

Reply:
I can't seem to get mypy to agree with me after several if-else guaranteeing the list will not be None, so I settled with `task_ids: List[Optional[str]]` and `task_config_ids: List[Optional[str]]`